### DISCUSSION TITLE
Early redirects to sign in page due to next-auth checking the session

### DISCUSSION THREAD

**Discussion by LGmatrix13** [nextauthjs/next-auth#3276] [Help] [answered]
So I have a sign page I redirect to if the user is not logged in. However, it sometimes happens before the session comes back. Is there anyway to prevent that or have it wait for the check? Thanks!

**Comment by balazsorban44** [comment, marked as answer]
have any code?

**Reply by LGmatrix13** [reply]
Yes! Sorry that I didn't add it originally....

So I have a `useEffect` in my `_app.js`

```
  useEffect(() => {
    if (!session && router.pathname == "/account") {
      router.push("/auth/login");
    }
  }, [session, router]);
```

**Reply by balazsorban44** [reply]
could you please be a bit more elaborate? maybe a reproduction would be the best

**Reply by LGmatrix13** [reply]
Sorry I didn't respond back, I was able to fix it using status in the useSession hook.
